### PR TITLE
[NFC] Apply small clang-tidy fixes

### PR DESCRIPTION
**Pull request by Keenuts**
- const for passed-by-value has no effect on prototypes.
- transitive includes.
- dead stores
- uses after move

### CODE REVIEW

File: test/enum_set_test.cpp [hunk 1]
```diff
@@ -841,24 +841,14 @@ TEST_P(CapabilitySetForEachTest, CopyConstructor) {
...
 }
 
-TEST_P(CapabilitySetForEachTest, MoveConstructor) {
-  // We need a writable copy to move from.
-  CapabilitySet copy(GetParam().capabilities);
-  CapabilitySet moved(std::move(copy));
-  EXPECT_THAT(ElementsIn(moved), Eq(GetParam().expected));
-
-  // The moved-from set is empty.
-  EXPECT_THAT(ElementsIn(copy), Eq(std::vector<spv::Capability>{}));
```
Comment (dneto0) [inline test/enum_set_test.cpp:851]: I think we should keep the test but delete line 851.

File: source/val/validate_memory.cpp [hunk 2]
```diff
@@ -785,7 +785,8 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
         (!_.HasCapability(spv::Capability::Float16) &&
          _.ContainsSizedIntOrFloatType(value_id, spv::Op::OpTypeFloat, 16))) {
       auto underlying_type = value_type;
-      while (underlying_type->opcode() == spv::Op::OpTypePointer) {
+      while (underlying_type &&
```
Comment (dneto0) [inline source/val/validate_memory.cpp:788]: This seems like a functional change, or at least not a clang-tidy change.  Is it already clear that value_type is not zero?
Should a new test be added for this?

I recommend pulling this part of the PR out to another. And at that point it's probably better to make value_type const and check it at its definition. at line 419 or so. That will save other guards in this function.

**Review by dneto0 — CHANGES_REQUESTED** [review]
Thanks for moving the ball forward.

File: source/val/validate_memory.cpp (hunk 2 above)
Comment (Keenuts) [reply source/val/validate_memory.cpp]: Yeah, felt weird to patch this, but those were indeed found through the static analyzer.
I reverted those, and I agree, some refactoring might be better.

**Comment by Keenuts** [comment]
rebased on main

File: test/enum_set_test.cpp (hunk 1 above)
Comment (dneto0) [reply test/enum_set_test.cpp:851]: Please put the test back, but delete line 851.

**Review by dneto0 — CHANGES_REQUESTED** [review]
Just one thing remaining: please put the test back.

**Review by s-perron — APPROVED** [review]

**Comment by sandratowry6** [comment]
I don't know how this work yet am new.

**Pull request merged**